lib/internal/streams/web.js

```diff
@@ -0,0 +1,4489 @@
+    if (!isReadableStream(readable)) {
+      throw new ERR_INVALID_ARG_TYPE(
+        'transform.readable',
+        'ReadableStream',
+        readable);
+    }
+    const writable = transform?.writable;
+    if (!isWritableStream(writable)) {
+      throw new ERR_INVALID_ARG_TYPE(
+        'transform.writable',
+        'WritableStream',
+        writable);
+    }
+
+    // The web platform tests require that these be handled one at a
+    // time and in a specific order. options can be null or undefined.
+    const preventAbort = options?.preventAbort;
+    const preventCancel = options?.preventCancel;
+    const preventClose = options?.preventClose;
+    const signal = options?.signal;
+
+    if (signal !== undefined && signal?.[kAborted] === undefined)
+      throw new ERR_INVALID_ARG_TYPE('options.signal', 'AbortSignal', signal);
+
+    if (isReadableStreamLocked(this))
+      throw new ERR_INVALID_STATE.TypeError('The ReadableStream is locked');
+    if (isWritableStreamLocked(writable))
+      throw new ERR_INVALID_STATE.TypeError('The WritableStream is locked');
+
+    const promise = readableStreamPipeTo(
+      this,
+      writable,
+      !!(preventClose),
+      !!(preventAbort),
+      !!(preventCancel),
+      signal);
+    setPromiseHandled(promise);
+
+    return readable;
+  }
+
+  /**
+   * @param {WritableStream} destination
+   * @param {StreamPipeOptions} [options]
+   * @returns {Promise<void>}
+   */
+  pipeTo(destination, options = {}) {
+    try {
+      if (!isReadableStream(this))
+        throw new ERR_INVALID_THIS('ReadableStream');
+      if (!isWritableStream(destination)) {
+        throw new ERR_INVALID_ARG_TYPE(
+          'transform.writable',
+          'WritableStream',
+          destination);
+      }
+
+      const preventAbort = options?.preventAbort;
+      const preventCancel = options?.preventCancel;
+      const preventClose = options?.preventClose;
+      const signal = options?.signal;
+
+      if (signal !== undefined && signal?.[kAborted] === undefined)
+        throw new ERR_INVALID_ARG_TYPE('options.signal', 'AbortSignal', signal);
+
+      if (isReadableStreamLocked(this))
+        throw new ERR_INVALID_STATE.TypeError('The ReadableStream is locked');
+      if (isWritableStreamLocked(destination))
+        throw new ERR_INVALID_STATE.TypeError('The WritableStream is locked');
+
+      return readableStreamPipeTo(
+        this,
+        destination,
+        !!(preventClose),
+        !!(preventAbort),
+        !!(preventCancel),
+        signal);
+    } catch (error) {
+      return PromiseReject(error);
+    }
+  }
+
+  /**
+   * @returns {ReadableStream[]}
+   */
+  tee() {
+    if (!isReadableStream(this))
+      throw new ERR_INVALID_THIS('ReadableStream');
+    return readableStreamTee(this, false);
+  }
+
+  // The values() method is not defined in the streams standard
+  // spec but it is called out in the section on async iteration.
+  // So... that's weird.
```

Comment:
The `values()` method comes from `async iterable<any>` on the `ReadableStream` Web IDL definition (step 5 of ["define the asynchronous iteration methods"](https://heycam.github.io/webidl/#define-the-asynchronous-iteration-methods)). 😉

Reply:
oh! right... I remember that now!

Btw, your typescript reference implementation has been absolutely invaluable as a reference point while going through the WPTs :-)